domainEntityFilters/www/filter.js

```diff
@@ -0,0 +1,74 @@
+//
+//  filter.js
+//
+//  Created by Zach Fox on 2019-06-07
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  See accompanying README.md for usage instructions.
+// 
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+
+function filterAdd(properties, originalProperties) {
+    if (properties.name.indexOf("Whiteboard") > -1) {
+        return properties;
+    }
+
+    return false;
+}
+
+
+function filterEdit(properties, originalProperties) {
+    if (originalProperties.name.indexOf("Whiteboard") > -1) {
+        return properties;
+    } else if (properties.privateUserData) {
+        var retProps = {
+            privateUserData: properties.privateUserData
+        }
+        return retProps;
+    }
+
+    return false;
+}
+
+
+function filterDelete(properties, originalProperties) {
+    if (originalProperties.name.indexOf("Whiteboard") > -1) {
+        return properties;
+    }
+
+    return false;
+}
+
+
+var FILTER_TYPE_ADD = 0;
+var FILTER_TYPE_EDIT = 1;
+var FILTER_TYPE_PHYSICS = 2;
+var FILTER_TYPE_DELETE = 3;
+function filter(properties, filterType, originalProperties) {
+    switch (filterType) {
+        case FILTER_TYPE_ADD:
+            return filterAdd(properties, originalProperties);
+        case FILTER_TYPE_EDIT:
+            return filterEdit(properties, originalProperties);
+        case FILTER_TYPE_PHYSICS:
+            return false;
+        case FILTER_TYPE_DELETE:
+            return filterDelete(properties, originalProperties);

```

Comment:
note above about original properties

Reply:
i disagree - `originalProperties` are necessary when deleting! (i tested this to verify)